tests/testthat/test-labelled.R

```diff
@@ -346,65 +346,68 @@ test_that("make_labelled returns expected skimr labels", {
   expected_skimr_labs <- list(
     field_name = "Variable / Field Name",
     field_label = "Field Label",
-    AsIs.n_unique = "Asis N Unique",
-    AsIs.min_length = "Asis Min Length",
-    AsIs.max_length = "Asis Max Length",
-    character.min = "Character Min",
-    character.max = "Character Max",
-    character.empty = "Character Empty",
-    character.n_unique = "Character N Unique",
-    character.whitespace = "Character Whitespace",
-    complex.mean = "Complex Mean",
-    Date.min = "Date Min",
-    Date.max = "Date Max",
-    Date.median = "Date Median",
-    Date.n_unique = "Date N Unique",
-    difftime.min = "Difftime Min",
-    difftime.max = "Difftime Max",
-    difftime.median = "Difftime Median",
-    difftime.n_unique = "Difftime N Unique",
-    factor.ordered = "Factor Ordered",
-    factor.n_unique = "Factor N Unique",
-    factor.top_counts = "Factor Top Counts",
-    haven_labelled.mean = "Haven Labelled Mean",
-    haven_labelled.sd = "Haven Labelled Sd",
-    haven_labelled.p0 = "Haven Labelled P0",
-    haven_labelled.p25 = "Haven Labelled P25",
-    haven_labelled.p50 = "Haven Labelled P50",
-    haven_labelled.p75 = "Haven Labelled P75",
-    haven_labelled.p100 = "Haven Labelled P100",
-    haven_labelled.hist = "Haven Labelled Hist",
-    list.n_unique = "List N Unique",
-    list.min_length = "List Min Length",
-    list.max_length = "List Max Length",
-    logical.mean = "Logical Mean",
-    logical.count = "Logical Count",
-    numeric.mean = "Numeric Mean",
-    numeric.sd = "Numeric Sd",
-    numeric.p0 = "Numeric P0",
-    numeric.p25 = "Numeric P25",
-    numeric.p50 = "Numeric P50",
-    numeric.p75 = "Numeric P75",
-    numeric.p100 = "Numeric P100",
-    numeric.hist = "Numeric Hist",
-    POSIXct.min = "Posixct Min",
-    POSIXct.max = "Posixct Max",
-    POSIXct.median = "Posixct Median",
-    POSIXct.n_unique = "Posixct N Unique",
-    Timespan.min = "Timespan Min",
-    Timespan.max = "Timespan Max",
-    Timespan.median = "Timespan Median",
-    Timespan.n_unique = "Timespan N Unique",
-    ts.start = "Ts Start",
-    ts.end = "Ts End",
-    ts.frequency = "Ts Frequency",
-    ts.deltat = "Ts Deltat",
-    ts.mean = "Ts Mean",
-    ts.sd = "Ts Sd",
-    ts.min = "Ts Min",
-    ts.max = "Ts Max",
-    ts.median = "Ts Median",
-    ts.line_graph = "Ts Line Graph"
+    skim_type = "Data Type",
```

Comment:
We should take a look at an actual Excel sheet that has these labels applied and decide how to trim these down (which data types never appear in REDCapTidieR? I think we will only ever have `character`, `Date`, `factor`, `logical`, `numeric`, and `POSIXct`. Correct? 

Can we look at the REDCap project that has all data types represented?

I think skimr also makes sparklines - for some reason I don't see this represented here?

For those that do, the ChatGPT generated labels are a good start but could be made better in many cases. For example, a better label for "0th Percentile of Numeric Values" is "Minimum".

Reply:
> We should take a look at an actual Excel sheet that has these labels applied and decide how to trim these down (which data types never appear in REDCapTidieR? I think we will only ever have `character`, `Date`, `factor`, `logical`, `numeric`, and `POSIXct`. Correct?

Technically you can also have a difftime come out from fields that are just clock values and not date times (expanded on below). This was causing an issue for us with the corrupted files because `openxlsx2` currently doesn't support difftimes (Jan Marvin e-mailed us about this).

> Can we look at the REDCap project that has all data types represented?

Using the REDCapTidieR Classic database we have two instruments that show this ("Data Field Types" & "Text Input Validation Types"), here's the results:

<details>
  <summary>Click me</summary>

```r
> out
# A tibble: 9 × 9
  redcap_form_name          redcap_form_label redcap_data redcap_metadata structure data_rows
  <chr>                     <chr>             <list>      <list>          <chr>         <int>
1 nonrepeated               Nonrepeated       <tibble>    <tibble>        nonrepea…         4
2 nonrepeated2              Nonrepeated2      <tibble>    <tibble>        nonrepea…         4
3 repeated                  Repeated          <tibble>    <tibble>        repeating         4
4 data_field_types          Data Field Types  <tibble>    <tibble>        nonrepea…         4
5 text_input_validation_ty… Text Input Valid… <tibble>    <tibble>        nonrepea…         4
6 api_no_access             API No Access     <tibble>    <tibble>        nonrepea…         4
7 api_no_access_2           API No Access 2   <tibble>    <tibble>        nonrepea…         4
8 survey                    Survey            <tibble>    <tibble>        nonrepea…         4
9 repeat_survey             Repeat Survey     <tibble>    <tibble>        repeating         3
# ℹ 3 more variables: data_cols <int>, data_size <lbstr_by>, data_na_pct <formttbl>
> sapply(out$redcap_data[[4]], class) %>% unique()
[1] "numeric"   "character" "factor"    "logical"  
> sapply(out$redcap_data[[5]], class) %>% unique()
[[1]]
[1] "numeric"

[[2]]
[1] "Date"

[[3]]
[1] "POSIXct" "POSIXt" 

[[4]]
[1] "character"

[[5]]
[1] "hms"      "difftime"

[[6]]
[1] "factor"
```

</details>

Results:
- numeric
- character
- factor
- logical
- Date
- POSIXct/POSIXt
- hms/difftime

Things we can probably get rid of:
- Haven
- Timespan
- Time Series 
- List
- AsIs? May want to keep for safety.

There may also be a different output for media uploads and signatures, but we aren't capturing those right now.

> I think skimr also makes sparklines - for some reason I don't see this represented here?

Anything with `*.hist` is the sparkline/histogram (which I think is just `haven.hist` and `numeric.hist`). It looks like it's also possible to get a `ts.linegraph`, but I'm not sure how we could invoke that so probably not something we want anyway.

> For those that do, the ChatGPT generated labels are a good start but could be made better in many cases. For example, a better label for "0th Percentile of Numeric Values" is "Minimum".

Sure, what others are you thinking? Change "100th Percentile of Numeric Values" to "Maximum"